Modules/Customer/Http/Controllers/CustomerController.php

```diff
@@ -121,26 +170,75 @@ public function update(UpdateCustomerRequest $request): JsonResponse
     }
 
     /**
-     * Remove the specified campaign from storage
+     * Update the specified customer in storage
      *
-     * @param string $customerKey
+     * @param UpdateCustomerInfoRequest $request
      * @return JsonResponse
      */
-    public function destroy(string $customerKey): JsonResponse
+    public function updateContactInfo(UpdateCustomerInfoRequest $request): JsonResponse
     {
         $user = auth()->user();
-        $customer = Customer::where('customer_key', $customerKey)->first();
+        $customer = Customer::where('customer_key', $request->customer_key)->first();
 
-        // return error if user not created the campaign
-        if ($user->cannot('delete', $customer)) {
+        // return error if user cannot update customer
+        if ($user->cannot('update', $customer)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->customerService->save($request->validated());
+
+        return response()->json($response);
+    }
+
+    /**
+     * Update the specified customer in storage
+     *
+     * @param UpdateCustomerShippingRequest $request
+     * @return JsonResponse
+     */
+    public function updateShippingDetails(UpdateCustomerShippingRequest $request): JsonResponse
+    {
+        $user = auth()->user();
+        $customer = Customer::where('customer_key', $request->customer_key)->first();
+
+        // return error if user cannot update customer
+        if ($user->cannot('update', $customer)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->customerService->save($request->validated());
+
+        return response()->json($response);
+    }
+
+    /**
+     * Remove the customers from storage
+     *
+     * @param DeleteCustomerRequest $request
+     * @return JsonResponse
+     */
+    public function destroy(DeleteCustomerRequest $request): JsonResponse
+    {
+        $user = auth()->user();
+
+        // return error if user is not a brand
+        if ($user->cannot('viewAny', Customer::class)) {
```

Comment:
@arijit1111 This should be `delete` Policy Method instead of `viewAny`

Reply:
It is only to check is the user brand or not, it is a list of customers id in request and now it validated as customers array

Reply:
it is only to check whether user is brand or not